batch7crud-roll1/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,216 @@
+	 */
+	@Override
+	protected void doGet(
+					HttpServletRequest request, HttpServletResponse response)
+					throws ServletException, IOException {
+
+		String requestInfo = request.getPathInfo();
+
+		if (requestInfo == null) {
+			list(request, response, 1);
+		}
+		else {
+
+			String[] pathParts = requestInfo.split("/");
+
+			if (CREATE.equals(pathParts[1])) {
+				create(request, response);
+			}
+			else if (EDIT.equals(pathParts[2])) {
+				int id = TypeCaster.toInt(pathParts[1]);
+				edit(request, response, id);
+			}
+		}
+	}
+
+	/**
+	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+	 */
+	@Override
+	protected void doPost(
+					HttpServletRequest request, HttpServletResponse response)
+					throws ServletException, IOException {
+
+		String requestInfo = request.getPathInfo();
+
+		String[] pathParts = requestInfo.split("/");
+
+		if (CREATE.equals(pathParts[1])) {
+			createProcess(request, response);
+		}
+		else if (EDIT.equals(pathParts[2])) {
+			int id = TypeCaster.toInt(pathParts[1]);
+			editProcess(request, response, id);
+		}
+		else if (DELETE.equals(pathParts[2])) {
+			int id = TypeCaster.toInt(pathParts[1]);
+			deleteProcess(request, response, id);
+		}
+	}
+
+	private void list(
+					HttpServletRequest request, HttpServletResponse response,
+					int page)
+					throws ServletException, IOException {
+
+		try {
+			List<Employee> employees = employeeService.fetch();
+			request.setAttribute("employees", employees);
+			request.getServletContext().getRequestDispatcher(
+							"/WEB-INF/views/employee/list.jsp").forward(
+							request, response);
+		}
+		catch (DataException e) {
+
+			logger.log(Level.SEVERE, e.getMessage());
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			request.setAttribute("message", e.getMessage());
+			request.getServletContext().getRequestDispatcher(
+							"/WEB-INF/views/error.jsp").forward(
+							request, response);
+		}
+	}
+
+	private void create(
+					HttpServletRequest request, HttpServletResponse response)
+					throws ServletException, IOException {
+
+		request.getServletContext().getRequestDispatcher(
+						"/WEB-INF/views/employee/create.jsp").forward(
+						request, response);
+	}
+
+	private void createProcess(
+					HttpServletRequest request, HttpServletResponse response)
+					throws ServletException, IOException {
+
+		int id = TypeCaster.toInt(request.getParameter("id"));
+		String userName = request.getParameter("userName");
+		String password = request.getParameter("password");
+		String fullName = request.getParameter("fullName");
+		String department = request.getParameter("department");
+		String address = request.getParameter("address");
+		Employee employee = new Employee(
+						id, userName, password, fullName, department, address);
+
+		try {
+			employeeService.create(employee);
+			response.sendRedirect(request.getContextPath() + "/employees");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/employees" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)